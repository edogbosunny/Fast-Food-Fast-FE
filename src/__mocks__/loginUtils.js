/* eslint-disable import/prefer-default-export */
export const props = {
  auth: {
    isAuthenticated: true,
    loading: null,
    user: { id: 1, iat: 1549439277, exp: 1549525677 },
  },
  errors: {
    isAuthenticated: false,
    loading: null,
    email: '',
    password: '',
    message: '',
  },
  user: {},
  history: {
    action: 'PUSH',
    push: jest.fn(),
  },
  location: {
    hash: '',
    pathname: '/login',
    search: '',
  },
  loginUser: jest.fn(),
  match: { path: '/login', url: '/login', isExact: true },
};

export const mockedLoginState = {
  auth: {
    isAuthenticated: true,
    loading: null,
    user: { id: 1, iat: 1549439277, exp: 1549525677 },
  },
  errors: {
    isAuthenticated: false,
    loading: null,
    email: '',
    password: '',
    message: '',
  },
};

export const nextProps = {
  auth: {
    isAuthenticated: true,
    loading: null,
    user: { id: 1, iat: 1549439277, exp: 1549525677 },
  },
  history: {
    action: 'PUSH',
    push: jest.fn(),
  },
};

export const Loginstate = {
  email: '',
  password: '',
};
