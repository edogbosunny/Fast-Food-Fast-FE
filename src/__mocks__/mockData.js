const mock = {
  loginDetails: {
    email: 'test@test.com',
    password: '123456',
  },
  registerDetails: {
    email: 'test1@test.com',
    password: 123456,
    confirmpassword: 123456,
    firstname: 'test',
    lastname: 'test,',
  },
  errRegisterDetails: {
    email: 'test1@test.com',
    password: 123456,
    confirmpassword: 123456,
    firstname: 'test',

  },
};
export default mock;
