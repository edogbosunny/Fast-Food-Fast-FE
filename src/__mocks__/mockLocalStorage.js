let localStorage = {};

export default {
  setItem(key, value) {
    return Object.assign(localStorage, { [key]: value });
  },
  removeItem(key) {
    return delete localStorage[key];
  },
  getItem(key) {
    return localStorage[key];
  },
  clear() {
    localStorage = {};
  },
};
