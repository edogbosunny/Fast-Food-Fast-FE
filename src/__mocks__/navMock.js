/* eslint-disable import/prefer-default-export */
export const props = {
  auth: {
    isAuthenticated: true,
    loading: null,
    user: { id: 1, iat: 1549439277, exp: 1549525677 },
  },
  errors: {
    isAuthenticated: false,
    loading: null,
    email: '',
    password: '',
    message: '',
  },
  history: {
    action: 'PUSH',
    push: jest.fn(),
  },
  location: {
    hash: '',
    pathname: '/login',
    search: '',
  },
  logoutUser: jest.fn(),
  match: { path: '/login', url: '/login', isExact: true },
};

export const mockState = {
  auth: {
    isAuthenticated: true,
    loading: null,
    user: { id: 1, iat: 1549439277, exp: 1549525677 },
  },
};
