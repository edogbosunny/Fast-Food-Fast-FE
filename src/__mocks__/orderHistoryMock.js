/* eslint-disable import/prefer-default-export */
export const props = {
  dispatch: jest.fn(),
  orderHistory: {
    loading: false,
    orderHistory: {
      count: 30,
      data: [{}],
      message: 'Food order retrieved succesfully',
    },
  },
};

export const tableProps = {
  getAllOrders: jest.fn(),
  orderHistory: {
    loading: false,
    orderHistory: {
      count: 30,
      data: [{}],
      message: 'Food order retrieved succesfully',
    },
  },
};
