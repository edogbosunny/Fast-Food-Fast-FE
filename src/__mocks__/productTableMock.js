/* eslint-disable import/prefer-default-export */
export const props = {
  cart: {
    count: 800,
    cart: {
      0: {
        meal_id: 1, price: '600', meal: 'Spaniish Noodles', qty: 1,
      },
    },
    items: null,
    cartCount: 800,
  },
  checkoutOrder: jest.fn(),
  order: {
    order: {},
  },
  history: {
    push: jest.fn(),
  },
  removeFromCart: jest.fn(),
};
