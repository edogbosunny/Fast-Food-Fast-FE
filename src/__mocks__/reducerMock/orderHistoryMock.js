const mock = {
  loading: true,
  orderHistory: {
    message: '',
    count: 0,
    data: [],
  },
};
export default mock;
