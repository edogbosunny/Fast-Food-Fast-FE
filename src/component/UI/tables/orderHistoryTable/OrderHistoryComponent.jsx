/* eslint-disable react/prop-types */
/* eslint-disable react/prefer-stateless-function */
import React, { Component } from 'react';
import { connect } from 'react-redux';

class OrderHistoryComponent extends Component {
  render() {
    const { orderHistory: { orderHistory } } = this.props;
    return orderHistory.data.map(orderedItems => (
      <tr className='tbg table-text-color tr' key={orderedItems.order_id}>
        <td className='td' data-label='orderid'>{orderedItems.order_id}</td>
        <td className='td' data-label='items'>
          {orderedItems.quantity}
          {' '}
          item(s)
          <button type='button' className='btn btn-success btn-sm' data-toggle="modal" data-target="#myModal"> View Order(s)</button>
          <button type='button' className='btn btn-danger btn-sm m-1'> Delete Order(s)</button>
        </td>
        <td className='td' data-label='cost'>
          â‚¦
          {orderedItems.cost}
        </td>
        <td className='td' data-label='date'>{orderedItems.created_on}</td>
        <td className='td' data-label='status'>{orderedItems.status}</td>
      </tr>
    ));
  }
}

const mapStateToProps = state => ({
  orderHistory: state.orderHistory,
});

export default connect(mapStateToProps)(OrderHistoryComponent);
