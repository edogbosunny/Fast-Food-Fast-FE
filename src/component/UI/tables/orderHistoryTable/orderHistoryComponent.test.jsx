import React from 'react';
import { shallow } from 'enzyme';
import { props } from '../../../../__mocks__/orderHistoryMock';
import { OrderHistoryComponent, mapStateToProps } from './OrderHistoryComponent';

describe('orderHistore component test-suit', () => {
  it('should match snapshot', () => {
    const wrapper = shallow(<OrderHistoryComponent {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should test mapStateToProps', () => {
    const state = mapStateToProps(mapStateToProps);
    expect(state).toBeTruthy();
  });
});
