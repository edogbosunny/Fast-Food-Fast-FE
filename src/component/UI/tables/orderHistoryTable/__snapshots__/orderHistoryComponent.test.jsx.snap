// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`orderHistore component test-suit should match snapshot 1`] = `
<tr
  className="tbg table-text-color tr"
>
  <td
    className="td"
    data-label="orderid"
  />
  <td
    className="td"
    data-label="items"
  >
     
    item(s)
    <button
      className="btn btn-success btn-sm"
      data-target="#myModal"
      data-toggle="modal"
      type="button"
    >
       View Order(s)
    </button>
    <button
      className="btn btn-danger btn-sm m-1"
      type="button"
    >
       Delete Order(s)
    </button>
  </td>
  <td
    className="td"
    data-label="cost"
  >
    â‚¦
  </td>
  <td
    className="td"
    data-label="date"
  />
  <td
    className="td"
    data-label="status"
  />
</tr>
`;
